packages/synapse-react-client/src/utils/functions/TableColumnSchemaUtils.ts

```diff
@@ -4,40 +4,67 @@ import {
   TableUpdateTransactionRequest,
 } from '@sage-bionetworks/synapse-types'
 import { SetOptional } from 'type-fest'
-import { isEqual } from 'lodash-es'
 import { createColumnModels } from '../../synapse-client'
+import { isEqualTreatUndefinedAsMissing } from './IsEqualTreatUndefinedAsMissing'
+
+function validateProposedSchema(
+  oldSchema: ColumnModel[],
+  proposedSchema: SetOptional<ColumnModel, 'id'>[],
+): void {
+  // If the proposed schema contains any IDs that are not in the old schema, throw an error
+  // We do this so we can accurately track updates to existing columns by using the ID that is already present
+  const oldColumnModelIds: Set<string> = new Set()
+  for (const columnModel of oldSchema) {
+    oldColumnModelIds.add(columnModel.id)
+  }
+  for (const columnModel of proposedSchema) {
+    if (columnModel.id != null && !oldColumnModelIds.has(columnModel.id)) {
+      throw new Error(
+        `Proposed schema contains a new column model with ID ${columnModel.id} that is not in the old schema.`,
+      )
+    }
+  }
+}
 
 /**
- * Creates the column models for a change request and returns
- * the change request to be passed to the Synapse backend.
- * @param accessToken
- * @param tableId
- * @param oldSchema
- * @param proposedSchema
+ * Creates the column models for a change request and returns the change request to be passed to the Synapse backend.
+ *
+ * Note that to update a column model and preserve table data, the ID of the
+ * @param accessToken the Synapse access token
+ * @param tableId the ID of the table to change
+ * @param oldSchema the current schema of the table
+ * @param proposedSchema the proposed schema of the table. For TableEntities, the IDs should be preserved for columns
+ *                       that should maintain their data, even if the column model has changed
  */
 export async function createTableUpdateTransactionRequest(
```

Comment:
@nickgros to create design ticket to address that users cannot discern between a modified column and a new column in the UI

Reply:
SWC-6666